lib/ansible/plugins/inventory/gcp_compute.py

```diff
@@ -0,0 +1,169 @@
+
+    def fetch_list(self, params, link, query):
+        '''
+            :param params: a dict containing all of the fields relevant to build URL
+            :param link: a formatted URL
+            :param query: a formatted query string
+            :return the JSON response containing a list of instances.
+        '''
+        module = GcpMockModule(params)
+        auth = GcpSession(module, 'compute')
+        response = auth.get(link, params={'filter': query})
+        return self.return_if_object(module, response)
+
+    def _get_query_options(self, config_data):
+        '''
+            :param config_data: contents of the inventory config file
+            :return A string
+        '''
+        queries = []
+        for obj_key in config_data['filters']:
+            for conditional in config_data['filters'][obj_key]:
+                if conditional not in ['ne', 'eq']:
+                    raise AnsibleParserError("All conditionals must be eq or ne")
+                for item in config_data['filters'][obj_key][conditional]:
+                    query = [obj_key, conditional, item]
+                    queries.append("(%s)" % ' '.join(query))
+
+        return ' '.join(queries)
+
+    def return_if_object(self, module, response):
+        '''
+            :param module: A GcpModule
+            :param response: A Requests response object
+            :return JSON response
+        '''
+        # If not found, return nothing.
+        if response.status_code == 404:
+            return None
+
+        # If no content, return nothing.
+        if response.status_code == 204:
+            return None
+
+        try:
+            response.raise_for_status
+            result = response.json()
+        except getattr(json.decoder, 'JSONDecodeError', ValueError) as inst:
+            module.fail_json(msg="Invalid JSON response with error: %s" % inst)
+        except GcpRequestException as inst:
+            module.fail_json(msg="Network error: %s" % inst)
+
+        if navigate_hash(result, ['error', 'errors']):
+            module.fail_json(msg=navigate_hash(result, ['error', 'errors']))
+        if result['kind'] != 'compute#instanceList':
+            module.fail_json(msg="Incorrect result: {kind}".format(**result))
+
+        return result
+
+    def parse(self, inventory, loader, path, cache=True):
+        super(InventoryModule, self).parse(inventory, loader, path)
+
+        config_data = self._validate_config(loader, path)
+
+        # get user specifications
+        if 'zones' not in config_data:
+            raise AnsibleParserError("Zones must be included in inventory YAML file")
+
+        # get user specifications
+        if 'project' not in config_data:
+            raise AnsibleParserError("Project must be included in inventory YAML file")
+
+        zones = config_data['zones']
+        query = self._get_query_options(config_data)
+
+        for zone in zones:
```

Comment:
Might be worth validating that zones is a list rather than a single zone provided as a string.